### ISSUE TITLE
ResolutionStrategy: failOnVersionConflict(): Conflict error message desperately needs improvement

### ISSUE THREAD

**Issue by eekboom** [gradle/gradle#5710]
tl;dr: When gradle detects a version conflict, it should show all transitive paths and versions to the dependency.

I am migrating a project from maven to gradle.

The build is configured to fail when there are different versions of a dependency in a configuration:
```
configurations.all {
	resolutionStrategy {
		// fail eagerly on version conflict (includes transitive dependencies)
		// e.g. multiple different versions of the same dependency (group and name are equal)
		failOnVersionConflict()
	}
}
```

I am expecting a helpful error message when a version conflict is detected, but all I get is a message like this:

> Could not resolve all dependencies for configuration ':app:compileClasspath'.
> > A conflict was found between the following modules:
>    - org.apache.logging.log4j:log4j-api:2.10.0
>    - org.apache.logging.log4j:log4j-api:2.11.0

This makes it very hard to figure out the problem. Image a build with a hundred modules and the version conflict deeply nested in transitive dependencies.

An example toy project that shows the dismal ghradle message is at https://github.com/eekboom/gradle-enforcer

I was expecting gradle to show all "paths" to these dependency including the version number.
That way it would be much, much easier to deal with the issue (for example by forcing a version or updating another dependency).

Compare the message generated by the maven enforcer plugin instead:

> Dependency convergence error for org.apache.commons:commons-lang3:3.6 paths to dependency are:
> +-com.acme.foo:foo-core-domain:105.0.0-SNAPSHOT
>   +-com.acme.foo:foo-common:105.0.0-SNAPSHOT
>     +-org.apache.commons:commons-lang3:3.6
> and
> +-com.acme.foo:foo-core-domain:105.0.0-SNAPSHOT
>   +-com.acme.foo:foo-common:105.0.0-SNAPSHOT
>     +-com.opencsv:opencsv:3.6
>       +-org.apache.commons:commons-lang3:3.6
> and
> +-com.acme.foo:foo-core-domain:105.0.0-SNAPSHOT
>   +-org.apache.commons:commons-lang3:3.5

**Comment by melix** [comment]
Hi @eekboom ,

You're right that the current message isn't nice, and the good news is that we already improved it on `master` (and an unmerged PR of mine). If you run it will now say:

```
Could not resolve all dependencies for configuration ':app:compileClasspath'.
> A conflict was found between the following modules:
    - org.apache.logging.log4j:log4j-api:2.10.0
    - org.apache.logging.log4j:log4j-api:2.11.0
  Run with:
      --scan or
      :app:dependencyInsight --configuration compileClasspath --dependency org.apache.logging.log4j:log4j-api
```

and then if you run `gradle :app:dependencyInsight --configuration compileClasspath --dependency org.apache.logging.log4j:log4j-api` it would show:

```
Dependency resolution failed because of conflicts between the following modules:
   - org.apache.logging.log4j:log4j-api:2.10.0
   - org.apache.logging.log4j:log4j-api:2.11.0

org.apache.logging.log4j:log4j-api:2.11.0
   variant "default+runtime" [
      org.gradle.status = release (not requested)
      Requested attributes not found in the selected variant:
         org.gradle.usage  = java-api
   ]
   Selection reasons:
      - Was requested
      - By conflict resolution : between versions 2.10.0 and 2.11.0

org.apache.logging.log4j:log4j-api:2.11.0
+--- project :modb
|    \--- compileClasspath
\--- org.apache.logging.log4j:log4j-core:2.11.0
     +--- project :moda
     |    \--- compileClasspath
     \--- project :modb (*)

org.apache.logging.log4j:log4j-api:2.10.0 -> 2.11.0
\--- project :moda
     \--- compileClasspath
```

Previously you couldn't even run the dependency insight report because of `failOnVersionConflict()`.

**Comment by melix** [comment]
This has beem merged into master.

**Comment by huehnerlady** [comment]
@melix  Hi, I like that you now can run dependency-insight. 
Would it be possible to run dependencyInsight for all faiiling things straight way - so you do not have to do that seperately for every dependency there is a conflict??

Happy to get help in writing a script for myself, but currently I wouldn't know how to get a list of conflicting dependencies
